app/ytdl.py

```diff
@@ -220,7 +227,15 @@ async def __add_entry(self, entry, quality, format, already):
             if not self.queue.exists(entry['id']):
                 dl = DownloadInfo(entry['id'], entry['title'], entry.get('webpage_url') or entry['url'], quality, format)
                 dldirectory = self.config.DOWNLOAD_DIR if (quality != 'audio' and format != 'mp3') else self.config.AUDIO_DOWNLOAD_DIR
-                self.queue.put(Download(dldirectory, self.config.OUTPUT_TEMPLATE, quality, format, self.config.YTDL_OPTIONS, dl))
+                output = self.config.OUTPUT_TEMPLATE
+                if "playlist" in entry:
+                    output = output.replace("%(playlist_index)s",entry["playlist_index"])
+                    output = output.replace("%(playlist_id)s",entry["playlist_id"])
+                    output = output.replace("%(playlist_title)s",entry["playlist_title"])
+                    output = output.replace("%(playlist_uploader)s",str(entry["playlist_uploader"]))
+                    output = output.replace("%(playlist_uploader_id)s",str(entry["playlist_uploader_id"]))
+                    output = output.replace("%(playlist)s",entry["playlist"])

```

Comment:
here we can write something generic that copes any playlist property that was supplied, so the code is more resilient to such changes:
```suggestion
                for property, value in entry.items():
                    if property.startswith("playlist"):
                        output = output.replace(f"%({property})s", value
```